src/uiSelectDirective.js

```diff
@@ -331,9 +338,6 @@ uis.directive('uiSelect',
               return;
             }
 
-            // Hide the dropdown so there is no flicker until $timeout is done executing.
-            dropdown[0].style.opacity = 0;
```

Comment:
Removing this introduces a problem when dropdown is positioned up

http://plnkr.co/edit/rM3rdaekwqG8wN77Hgb6?p=preview

Enter text in select2 "up" and delete with `backspace` quickly (I typed `ada` to get most results).

With this line removed the select gets shifted/overlayed with the list as a flicker. (Comment/uncomment line ~1261)

We need to detect if dropdown is positioned up and apply this along with your additions on line 298.